atlp-rwanda/e-comm-team-amigos-bn
# #184581179 Two Factor Authentication for Sellers

#### What does this PR do?

This PR verify if user is a seller and he/she can login as a seller

#### Description of Task to be completed?

- [x] A seller will be successfully login with email and password

- [x] A seller should receive email containing random number for 2FA to verify if he/she is owner of stall.

- [x] Document feature using swagger.


#### How should this be manually tested?
- Clone this repo `git clone https://github.com/atlp-rwanda/e-comm-team-amigos-bn`
- Switch your working directory `cd e-comm-team-amigos-bn`
- Create an` .env` file and update it with `.env-example`
- Install all the dependencies by running` npm install`
- Then run the project by typing `npm dev` to start server
- Make sure that postman is installed, open postman or any other API testing tool of your choice
- choose route to be `POST`
- Navigate `localhost:3000/user/login`
- Make sure you created account, enter your username and password
- if you are a seller you will receive email containing `OTP (one-time-password) `
- you can use it to successfully login

#### Endpoints to be tested:

- `POST user/login`

#### Any background context you want to provide?

- PostgreSQL installed in your local machine
- Node JS installed in your local machine

#### What are the relevant pivotal tracker/Trello stories?

Story ID: [#184581179](https://www.pivotaltracker.com/story/show/184581179)

#### Screenshot of  `swagger documentation`  ,  `test`   and  `email format`  to be sent

![Screenshot (134)](https://user-images.githubusercontent.com/79166151/224707737-2312e089-6827-48e7-9858-0525082d4f77.png)

![Screenshot (133)](https://user-images.githubusercontent.com/79166151/224707782-deb469c2-16d5-4ccc-b916-8384da00b692.png)

![email](https://user-images.githubusercontent.com/79166151/224961520-c8598d7f-a430-4d47-8426-ebbb5cf14b4c.png)
